redcap/project.py

```diff
@@ -325,7 +327,11 @@ def export_records(self, records=None, fields=None, forms=None,
                 else:
                     df_kwargs = {'index_col': self.def_field}
             buf = StringIO(response)
-            df = read_csv(buf, **df_kwargs)
+            df = []
+            if type == 'flat':
```

Comment:
I think `df_kwargs` should be honored even on `type=eav` requests?

Reply:
Here is the error I encountered when format=df and type=eav and I think it is because df_kwargs is not structured correctly when type=eav. And it won't have this error when format!=df.

 File "C:/Users/shiclab/PyCap/sync_data_eav.py", line 29, in sync_data
    data_frame = project.export_records(format='df', type='eav')

  File "redcap\project.py", line 331, in export_records
    df = read_csv(buf, **df_kwargs)

  File "C:\ProgramData\Anaconda2\lib\site-packages\pandas\io\parsers.py", line 655, in parser_f
    return _read(filepath_or_buffer, kwds)

  File "C:\ProgramData\Anaconda2\lib\site-packages\pandas\io\parsers.py", line 411, in _read
    data = parser.read(nrows)

  File "C:\ProgramData\Anaconda2\lib\site-packages\pandas\io\parsers.py", line 982, in read
    ret = self._engine.read(nrows)

  File "C:\ProgramData\Anaconda2\lib\site-packages\pandas\io\parsers.py", line 1797, in read
    index, names = self._make_index(data, alldata, names)

  File "C:\ProgramData\Anaconda2\lib\site-packages\pandas\io\parsers.py", line 1312, in _make_index
    index = self._get_simple_index(alldata, columns)

  File "C:\ProgramData\Anaconda2\lib\site-packages\pandas\io\parsers.py", line 1346, in _get_simple_index
    i = ix(idx)

  File "C:\ProgramData\Anaconda2\lib\site-packages\pandas\io\parsers.py", line 1340, in ix
    raise ValueError('Index %s invalid' % col)

ValueError: Index src_subject_id invalid

Reply:
Oh I see, the issue is just above, if `df_kwargs` are not passed then it defaults to specifying `index_col` in `df_kwargs`. What I think should happen is that `df_kwargs` should be used, but  this block

https://github.com/yawenl/PyCap/blob/3064ae4dc247d6b6614cb9a4472d17e88dfbc9cd/redcap/project.py#L322-L328

should include a if-branch for determining the proper index when `type=eav`?

Reply:
I just changed the code:
when type=eav, if `df_kwargs ` is null, I set it to empty dict. I'm not so sure what proper index it should be set to other than empty dict... Do you have an idea? Currently setting it to empty dict works for my project in RedCap.